InvenTree/company/migrations/0035_supplierpart_update.py

```diff
@@ -0,0 +1,95 @@
+
+            if supplier_part.manufacturer_part:
+                print(f'[ERROR: MANUFACTURER PART ALREADY EXISTS]')
+                continue
+
+            part = supplier_part.part
+            if not part:
+                print(f'[ERROR: SUPPLIER PART IS NOT CONNECTED TO PART]')
+                continue
+            
+            manufacturer = supplier_part.manufacturer
+            MPN = supplier_part.MPN
+            link = supplier_part.link
+            description = supplier_part.description
+
+            if manufacturer or MPN:
+                print(f' | {part.name[:15].ljust(15)}', end='')
+                
+                try:
+                    print(f' | {manufacturer.name[:15].ljust(15)}', end='')
+                except AttributeError:
+                    print(f' | {"EMPTY MANUF".ljust(15)}', end='')
+
+                try:
+                    print(f' | {MPN[:15].ljust(15)}', end='')
+                except TypeError:
+                    print(f' | {"EMPTY MPN".ljust(15)}', end='')
+
+                print('\t', end='')
+
+                # Create ManufacturerPart
+                manufacturer_part = ManufacturerPart(part=part, manufacturer=manufacturer, MPN=MPN, description=description, link=link)
+                created = False
+                try:
+                    with transaction.atomic():
+                        manufacturer_part.save()
+                    created = True
+                except IntegrityError:
+                    manufacturer_part = ManufacturerPart.objects.get(part=part, manufacturer=manufacturer, MPN=MPN)
+
+                # Link it to SupplierPart
+                supplier_part.manufacturer_part = manufacturer_part
+                supplier_part.save()
+
+                if created:
+                    print(f'[SUCCESS: MANUFACTURER PART CREATED]')
+                else:
+                    print(f'[IGNORED: MANUFACTURER PART ALREADY EXISTS]')
+            else:
+                print(f'[IGNORED: MISSING MANUFACTURER DATA]')
+
+        print(f'{"-"*10}\nDone\n')
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('company', '0033_auto_20210410_1528'),
+        ('company', '0034_manufacturerpart'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='supplierpart',
+            name='manufacturer_part',
+            field=models.ForeignKey(blank=True, help_text='Select manufacturer part', null=True, on_delete=django.db.models.deletion.CASCADE, related_name='supplier_parts', to='company.ManufacturerPart', verbose_name='Manufacturer Part'),
+        ),
+        # Make new ManufacturerPart with SupplierPart "manufacturer" and "MPN"
+        # fields, then link it to the new SupplierPart "manufacturer_part" field
+        migrations.RunPython(supplierpart_make_manufacturer_parts),
+        migrations.RemoveField(
+            model_name='supplierpart',
+            name='MPN',
+        ),
+        migrations.RemoveField(
```

Comment:
In general it is not recommended to combine data migrations and schema migrations. 

Instead, this migration file should be split into three separate files:

1. Add the field `manufacturer_part`
2. Perform data migration
3. Remove the fields `MPN` and `manufacturer`

Reply:
Split SupplierPart object update in 3 files, as recommended :+1: